wordpress-mobile/WordPress-Android
# [SotW 2023] Implement card analytics

Part of #19709 

Implement analytics for the card actions, following the spec in the issue above.

#### Card shown
`sotw_2023_nudge_post_event_card_shown`

#### Card hidden ("hide this" button tapped)
`sotw_2023_nudge_post_event_card_hide_tapped`

#### CTA tapped
`sotw_2023_nudge_post_event_card_cta_tapped`

-----

## To Test:

1. Install and log into the WordPress app
2. Set up device/app in a way that SotW card is displayed (more info: https://github.com/wordpress-mobile/WordPress-Android/pull/19718#issue-2021569561)
3. Go to `My Site`
4. **Verify** in app logs: `🔵 Tracked: sotw_2023_nudge_post_event_card_shown`
5. Tap the "Watch now" button
6. **Verify** in app logs: `🔵 Tracked: sotw_2023_nudge_post_event_card_cta_tapped`
7. Tap the "Hide this" button
8. **Verify** in app logs: `🔵 Tracked: sotw_2023_nudge_post_event_card_hide_tapped`

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

9. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

10. What automated tests I added (or what prevented me from doing so)

    - Added unit tests for tracking

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
No UI Changes.
